contracts/VoucherHub.sol

```diff
@@ -253,6 +253,40 @@ contract VoucherHub is
         emit VoucherRefunded(msg.sender, amount);
     }
 
+    /**
+     * Drain funds from voucher if it is expired. Transfer all RLC balance
+     * on PoCo from voucher to voucherHub and burn all credits.
+     * @param voucher address of the expired voucher to drain
+     */
+    function drainVoucher(address voucher) external {
+        VoucherHubStorage storage $ = _getVoucherHubStorage();
+        require($._isVoucher[voucher], "VoucherHub: unknown voucher");
```

Comment:
```suggestion
```

Reply:
It would consume gas for nothing.

Reply:
And what happens if you drain twice?

Reply:
The same thing would happen, but I don't see the link. We can check if the balance is not null to avoid consuming gas but it's not an argument to remove the check here.

Reply:
To be explicit (!), I was suggesting:
```
    function drainVoucher(address voucher) external {
        uint256 amount = balanceOf(voucher);
        require(amount > 0, "VoucherHub: nothing to drain");
        _burn(voucher, amount);
        emit VoucherDrained(voucher, amount);
        Voucher(voucher).drain(amount);
    }
```

Reply:
Indeed, your original suggestion wasn't as explicit to me.
We do agree on this version, thanks :+1:
 [Check balance before drain](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/25/commits/5a137b107b2f3ef2fa4e2d0881c660777473227c)